code/modules/mob/living/silicon/robot/drone/drone.dm

```diff
@@ -457,13 +460,16 @@
 /mob/living/silicon/robot/drone/construction/process_level_restrictions()
 	//Abort if they should not get blown
 	if(lock_charge || scrambled_codes || emagged)
-		return
-	//Check if they are not on station level -> abort
+		return FALSE
+	//Check if they are not on a station level -> else abort
 	var/turf/T = get_turf(src)
-	if(!T || isNotStationLevel(T.z))
```

Comment:
Also fixes this inverted check here, not sure what was going on with that.

Reply:
That check was there so construction drones that might exist at the away site don't blow up

Reply:
Okay. I've changed it so that Construction Drones may go wherever. I don't think they should blow up if they come to the station.

Reply:
Okay... I changed it again so Construction Drones must stay on areas connected to where they are initialized, or else boom.